core/src/include/ModelConfig.hpp

```diff
@@ -0,0 +1,47 @@
+/*!
+ * @file ModelConfig.hpp
+ *
+ * @date 14 Apr 2023
+ * @author Tim Spain <timothy.spain@nersc.no>
+ */
+
+#ifndef MODELCONFIG_HPP
+#define MODELCONFIG_HPP
+
+#include "include/ModelMetadata.hpp"
+#include "include/Time.hpp"
+
+#include <map>
+#include <string>
+
+namespace Nextsim {
+
+class Model;
+
+class ModelConfig {
+public:
+    const static std::map<int, std::string> keyMap;
+    static ConfigMap getConfig();
+
+    enum {
+        DUMMY_KEY,
```

Comment:
What is the dummy key for...?

Reply:
Notice that the configuration keys in `Model.hpp` are assigned values from this class. This is a rather unsatisfactory method, as adding the MPI partition key breaks this system. I have added some comments to explain this, but also moved the MPI configuration key.